platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/streams/SerializableDataInputStream.java

```diff
@@ -409,41 +502,59 @@ public <T extends SelfSerializable> T[] readSerializableArray(
     /**
      * Read an array of serializable objects from the stream.
      *
-     * @param arrayConstructor
-     * 		a method that returns an array of the requested size
-     * @param maxListSize
-     * 		maximal number of object should read
-     * @param readClassId
-     * 		set to true if the class ID was written to the stream
-     * @param serializableConstructor
-     * 		an object that returns new instances of the class
-     * @param <T>
-     * 		the implementation of {@link SelfSerializable} used
+     * @param arrayConstructor        a method that returns an array of the requested size
+     * @param maxListSize             maximal number of object should read
+     * @param readClassId             set to true if the class ID was written to the stream
+     * @param permissibleCLassIds     a set of class IDs that are allowed to be read, will throw an IOException if asked
+     *                                to deserialize a class not in this, all class IDs are permitted if null. Ignored
+     *                                if readClassId is false.
+     * @param serializableConstructor an object that returns new instances of the class
+     * @param <T>                     the implementation of {@link SelfSerializable} used
      * @return An array of the instances of the class previously written
-     * @throws IOException
-     * 		thrown if any IO problems occur
+     * @throws IOException thrown if any IO problems occur
      */
     public <T extends SelfSerializable> T[] readSerializableArray(
-            final IntFunction<T[]> arrayConstructor,
+            @NonNull final IntFunction<T[]> arrayConstructor,
             final int maxListSize,
             final boolean readClassId,
-            final Supplier<T> serializableConstructor)
+            @NonNull final Supplier<T> serializableConstructor,
+            @Nullable final Set<Long> permissibleCLassIds)
             throws IOException {
 
-        final List<T> list = readSerializableList(maxListSize, readClassId, id -> serializableConstructor.get());
+        final List<T> list = readSerializableList(
+                maxListSize, readClassId, id -> serializableConstructor.get(), permissibleCLassIds);
         if (list == null) {
             return null;
         }
         return list.toArray(arrayConstructor.apply(list.size()));
     }
 
+    /**
+     * Read an array of serializable objects from the stream.
+     *
+     * @param arrayConstructor        a method that returns an array of the requested size
+     * @param maxListSize             maximal number of object should read
```

Comment:
grammar: "maximal number of object should read"

Reply:
fixed